.github/renovate.json5

```diff
@@ -0,0 +1,134 @@
+
+  // Give ecosystem time to catch up.
+  // npm allows maintainers to unpublish a release up to 3 days later.
+  // https://docs.renovatebot.com/configuration-options/#minimumreleaseage
+  "minimumReleaseAge": "3", 
+
+  // Create PRs, but do not update them without manual action.
+  // Reduces spurious retesting in repositories that have many PRs at a time.
+  // https://docs.renovatebot.com/configuration-options/#rebasewhen
+  "rebaseWhen": "never",
+
+  // Organizational processes.
+  // https://docs.renovatebot.com/configuration-options/#dependencydashboardlabels
+  "dependencyDashboardLabels": [
+    "type: process",
+  ],
+  // https://docs.renovatebot.com/configuration-options/#prbodynotes
+  "prBodyNotes": [
+    "Reviewer is responsible for dependency update. Ensure adequate automated",
+    "or manual testing is performed before merge.",
+  ],
+
+  "packageRules": [
+    // Report impact of a dependency update via PR label.
+    {
+      "matchUpdateTypes": ["patch"],
+       "labels": ["semver: patch"],
+    },
+    {
+      "matchUpdateTypes": ["minor"],
+       "labels": ["semver: minor"],
+    },
+    {
+      "matchUpdateTypes": ["major"],
+       "labels": ["semver: major"],
+    },
+
+    // Language-specific behaviors.
+    {
+      "matchLanguages": ["go"],
+      "postUpdateOptions": ["gomodTidy"],
+    },
+
+    // Tooling & Runtime behaviors.
+    {
+      // Covers Dockerfiles, cloudbuild.yaml, and other docker-based tools.
+      "groupName": "Docker Runtimes",
+      "matchDatasources": ["docker"],
+      // TODO: Uncomment if your Dockerfiles are not included in samples.
+      // Increases build repeatability, image cache use, and supply chain security.
+      // "pinDigests": true,
+    },
+    {
+       "groupName": "GitHub Actions",
+       "matchManagers": ["github-actions"],
+       "pinDigests": true,
+    },
+
+    // Language groupings.
+    // These are separated to facilitate PR review by single-language experts.
+    {
+      "groupName": "go-nonmajor",
+      "matchLanguages": ["go"],
+      "matchUpdateTypes": ["minor", "patch"],
+    },
+    {
+      "groupName": "python-nonmajor",
+      "matchLanguages": ["python"],
+      "matchUpdateTypes": ["minor", "patch"],
+    },
+    {
+      "groupName": "java-nonmajor",
+      "matchLanguages": ["java"],
+      "matchUpdateTypes": ["minor", "patch"],
+    },
+    {
+      "groupName": "js-nonmajor",
+      "matchLanguages": ["js"],
+      "matchUpdateTypes": ["minor", "patch"],
+    },
+    {
+      "groupName": "nodejs-nonmajor",
+      "matchLanguages": ["node"],
+      "matchUpdateTypes": ["minor", "patch"],
+    },
+    {
+      "groupName": "npm-dev",
+      "matchManagers": ["npm"],
+      "matchDepTypes": ["devDependencies"],
+      "separateMajorMinor": false,
+    },
+    {
+      "groupName": "Terraform",
+      "matchManagers": ["terraform"],
+      "separateMajorMinor": false,
+    },
+
+    // Updating coupled ecosystems.
```

Comment:
Should we have a per language renovate or do we want to start collecting more packages here?

Reply:
My original review had the idea we would port some learns from here to the mono repos. Now I am declaring that a secondary goal.